snuba/api/query.py

```diff
@@ -88,7 +94,7 @@ def raw_query(
     with state.deduper(query_id if use_deduper else None) as is_dupe:
         timer.mark("dedupe_wait")
 
-        result = state.get_result(query_id) if use_cache else None
+        result = cache.get(query_id) if use_cache else None
```

Comment:
This becomes `Union[Any, None]` which is better than `Any`, I guess.

As noted in the description, I am going to follow up with a change that changes this `Cache[Any]` to `Cache[Result]` but that's going to take a bit more work.

I also would like to remove the `use_cache` flag (at least for testing) to ensure that results can be serialized to the cache since this has bit us before on deployments.

Reply:
Please keep me informed when you are planning to make these changes. I was planning to provide some more nuanced way to enable/disble caching (for long discovery queries), like caching requests longer than "x" msec

Reply:
Sounds good, will do. I don't expect to do anything else here in the near term, other than possibly put together that `Codec[str, snuba.reader.Result]` object, but that might get back burnered based on how much wrangling I have to do with stats and whatever in the rest of this function. Don't let me block your progress.

One thing to consider (if you haven't already): I don't think this is the right place to do the kind of caching you described. We should probably be implementing the prior to splitting, I think. I'd expect this abstraction to be useful there as well (and probably replace the cache instance here.)